tools/evmc/main.cpp

```diff
@@ -39,89 +39,95 @@ struct HexValidator : public CLI::Validator
 };
 }  // namespace
 
-int main(int argc, const char** argv)
+int main(int argc, const char** argv) noexcept
 {
     using namespace evmc;
 
-    static HexValidator Hex;
-
-    std::string vm_config;
-    std::string code_arg;
-    int64_t gas = 1000000;
-    auto rev = EVMC_LATEST_STABLE_REVISION;
-    std::string input_arg;
-    auto create = false;
-    auto bench = false;
-
-    CLI::App app{"EVMC tool"};
-    const auto& version_flag = *app.add_flag("--version", "Print version information and exit");
-    const auto& vm_option =
-        *app.add_option("--vm", vm_config, "EVMC VM module")->envname("EVMC_VM");
-
-    auto& run_cmd = *app.add_subcommand("run", "Execute EVM bytecode")->fallthrough();
-    run_cmd.add_option("code", code_arg, "Bytecode")->required()->check(Hex | CLI::ExistingFile);
-    run_cmd.add_option("--gas", gas, "Execution gas limit", true)->check(CLI::Range(0, 1000000000));
-    run_cmd.add_option("--rev", rev, "EVM revision", true);
-    run_cmd.add_option("--input", input_arg, "Input bytes")->check(Hex | CLI::ExistingFile);
-    run_cmd.add_flag(
-        "--create", create,
-        "Create new contract out of the code and then execute this contract with the input");
-    run_cmd.add_flag(
-        "--bench", bench,
-        "Benchmark execution time (state modification may result in unexpected behaviour)");
-
     try
     {
-        app.parse(argc, argv);
-
-        evmc::VM vm;
-        if (vm_option.count() != 0)
+        HexValidator Hex;
+
+        std::string vm_config;
+        std::string code_arg;
+        int64_t gas = 1000000;
+        auto rev = EVMC_LATEST_STABLE_REVISION;
+        std::string input_arg;
+        auto create = false;
+        auto bench = false;
+
+        CLI::App app{"EVMC tool"};
+        const auto& version_flag = *app.add_flag("--version", "Print version information and exit");
+        const auto& vm_option =
+            *app.add_option("--vm", vm_config, "EVMC VM module")->envname("EVMC_VM");
+
+        auto& run_cmd = *app.add_subcommand("run", "Execute EVM bytecode")->fallthrough();
+        run_cmd.add_option("code", code_arg, "Bytecode")
+            ->required()
+            ->check(Hex | CLI::ExistingFile);
+        run_cmd.add_option("--gas", gas, "Execution gas limit", true)
+            ->check(CLI::Range(0, 1000000000));
+        run_cmd.add_option("--rev", rev, "EVM revision", true);
+        run_cmd.add_option("--input", input_arg, "Input bytes")->check(Hex | CLI::ExistingFile);
+        run_cmd.add_flag(
+            "--create", create,
+            "Create new contract out of the code and then execute this contract with the input");
+        run_cmd.add_flag(
+            "--bench", bench,
+            "Benchmark execution time (state modification may result in unexpected behaviour)");
+
+        try
```

Comment:
Why do you need to nest it, seems it could be a single outer try-block for everything?

Reply:
In the inner `try-catch`, in the `catch (const CLI::ParseError& e)` the `app` object is used.